agent/consul/streaming_endpoint.go

```diff
@@ -128,11 +128,8 @@ func sendEvent(event stream.Event, aclFilter *aclFilter, eval *bexpr.Evaluator,
 	// hash it to save space. This is only used to determine if an object has been
 	// removed and needs to be deleted from the client's cache so we make the tradeoff
 	// of using less memory here at the extremely small risk of a hash collision.
-	rawId := event.ID()
-	hash, err := blake2b.New256(nil)
-	if err != nil {
-		return err
-	}
+	rawId := event.ContentID()
+	hash := fnv.New32a()
 	hash.Write([]byte(rawId))
 	bytes := hash.Sum(nil)
 	idHash := string(bytes)
```

Comment:
If we are going to the bother of a 32 bit hash to save space it's wasteful to convert it back into a string again and have the overhead of another string allocation and string map (which implies pointers and size counters per key etc). This string will also be invalid UTF-8 since the bytes are essentially random.

We should make the map be keyed by `uint32` and then use `hash.Uint32()` to generate the key.